src/hera/_cli/generate/hera.py

```diff
@@ -0,0 +1,434 @@
+                hera_imports.extend(sub_hera_imports)
+                model_imports.extend(sub_model_imports)
+                hera_template_str.append(f"{indent}{val}")
+            indent = " " * 8
+    elif hera_template_class == DAG and template.dag:
+        for task in template.dag.tasks:
+            val, sub_hera_imports, sub_model_imports = convert_to_hera_equivalent(task, Task)
+            hera_imports.extend(sub_hera_imports)
+            model_imports.extend(sub_model_imports)
+            hera_template_str.append(f"{indent}{val}")
+    elif hera_template_class == ContainerSet and template.container_set:
+        for container in template.container_set.containers:
+            val, sub_hera_imports, sub_model_imports = convert_to_hera_equivalent(container, ContainerNode)
+            hera_imports.extend(sub_hera_imports)
+            model_imports.extend(sub_model_imports)
+            hera_template_str.append(f"{indent}{val}")
+
+    return (
+        "\n".join(hera_template_str),
+        hera_imports,
+        model_imports,
+    )
+
+
+def template_to_python(template: Template) -> Tuple[str, List[str], List[str]]:
+    if template.container is not None:
+        return convert_to_hera_template(template, template.container, Container)
+    elif template.script is not None:
+        return convert_to_hera_template(template, template.script, Script)
+    elif template.http is not None:
+        return convert_to_hera_template(template, template.http, HTTP)
+    elif template.data is not None:
+        return convert_to_hera_template(template, template.data, Data)
+    elif template.resource is not None:
+        return convert_to_hera_template(template, template.resource, Resource)
+    elif template.suspend is not None:
+        return convert_to_hera_template(template, template.suspend, Suspend)
+    elif template.dag is not None:
+        return convert_to_hera_invocator_template(template, DAG, template.dag)
+    elif template.steps is not None:
+        return convert_to_hera_invocator_template(template, Steps)
+    elif template.container_set is not None:
+        return convert_to_hera_invocator_template(template, ContainerSet, template.container_set)
+    else:
+        # No-op (convert Template to itself) and rearrange imports
+        val, hera_imports, model_imports = convert_to_hera_equivalent(template, Template)
+        hera_imports.remove("Template")
+        model_imports.append("Template")
+        return val, hera_imports, model_imports
+
+
+def model_to_python(model: BaseModel) -> Tuple[str, List[str], List[str]]:
+    model_name = model.__class__.__name__
+    if isinstance(model, Template):
+        return template_to_python(model)
+
+    hera_imports = []
+    model_imports = [model_name]
+    class_def = [f"{model_name}("]
+    for attr in ChainMap(*(get_annotations(c) for c in model.__class__.__mro__)):
+        if attr == "__slots__" or getattr(model, attr) is None:
+            continue
```

Comment:
is there not a Pydantic API exposing this? what is the difference from `__fields__`?

Reply:
I think this was a mindless copy-paste from the `ModelMapperMixin` 🤔 (And now I'm trying to remember why we had to use `__mro__` in the first place...).

Changed to `__fields__` and all tests still pass so 👍 from me